addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -674,7 +674,16 @@ def write(self, values):
         employee_id = values.get('employee_id', False)
         if values.get('state'):
             self._check_approval_update(values['state'])
+
         result = super(HolidaysAllocation, self).write(values)
+
+        # Check constraints after write to consider updated values.
+        if 'number_of_days_display' in values or 'number_of_hours_display' in values:
+            consumed_leaves = self.employee_id._get_consumed_leaves(leave_types=self.holiday_status_id)
+            for allocation in self:
+                if dict(consumed_leaves[1])[allocation.employee_id][allocation.holiday_status_id]['excess_days']:
```

Comment:
As said in DM, I think it would be best to get the values both before and after the write and compare differences, because without that it will raise the error for employees with historical issues and pre-existing excess days.
Before the `super().write()` add something like
```py
previous_consumed_leaves = self.employee_id._get_consumed_leaves(leave_types=self.holiday_status_id)
```
and raise the error if the content between the respective excess_days has changed 🤔